Content.Server/Connection/ConnectionManager.IPIntel.cs

```diff
@@ -0,0 +1,226 @@
+        {
+            return (false, string.Empty);
+        }
+
+        // Check our cache
+        var ip = e.IP.Address;
+        var query = await _db.GetIPIntelCache(ip);
+        var expired = false;
+
+        // Does it exist?
+        if (query.Count != 0)
+        {
+            // Skip to score check if result is older than _cacheDays
+            if (!query.Any(a => (DateTime.Now - a.Time).TotalDays > _cacheDays))
+            {
+                var cachedScore = query.FirstOrDefault()?.Score.ToString(CultureInfo.CurrentCulture);
+                return await ScoreCheck(cachedScore!, e, false, false);
+            }
+            // This record is expired and should be updated.
+            expired = true;
+        }
+
+        // Check our api limits. If we are ratelimited we back out.
+        await HandleRatelimit();
+
+        if (_ratelimitedMinute || _ratelimitedDay)
+            return _rejectLimited ? (true, Loc.GetString("ipintel-server-ratelimited")) : (false, string.Empty);
+
+        // Ensure our contact email is good to use.
+        if (string.IsNullOrEmpty(_contactEmail) || !_contactEmail.Contains('@') || !_contactEmail.Contains('.'))
+        {
+            _sawmill.Error("IPIntel is enabled, but contact email is empty or not a valid email, treating this connection like an unknown IPIntel response.");
+            return _rejectUnknown ? (true, Loc.GetString("generic-misconfigured")) : (false, string.Empty);
+        }
+
+        _currentRequestsDay++;
+        _currentRequestsMinute++;
+        // Info about flag B: https://getipintel.net/free-proxy-vpn-tor-detection-api/#flagsb
+        // TLDR: We don't care about knowing if a connection is compromised.
+        // We just want to know if it's a vpn. This also speeds up the request by quite a bit. (A full scan can take 200ms to 5 seconds. This will take at most 120ms)
+        using var request = await _httpClient.GetAsync($"{_baseUrl}/check.php?ip={ip}&contact={_contactEmail}&flags={_flags}");
+
+        if (request.StatusCode == HttpStatusCode.TooManyRequests)
+        {
+            // TODO: Ok so nik did kinda kick start my brain that... well... this could either be a daily limit or a minute limit and i would have no idea...
+            // I was gonna treat this as daily but yeah i see the issue here...
+            // We are just gonna try to estimate our ratelimit
+            //... maybe wait a minute send the request again and if we are still ratelimited assume we hit the daily limit and cry
+            // also helps to have minute/day requests stored in the db
+
+            _sawmill.Warning("We hit the IPIntel request limit at some point.");
+            return _rejectLimited ? (true, Loc.GetString("ipintel-server-ratelimited")) : (false, string.Empty);
+        }
+
+        var response = await request.Content.ReadAsStringAsync();
+
+        switch (response)
```

Comment:
You should check whether the status code is 400 so you can handle any new future error codes properly.